src/modules/evaluateExpression/evaluateExpressionTestData.ts

```diff
@@ -0,0 +1,708 @@
+};
+
+testData.organisation = {
+  id: 1,
+  name: 'XYZ Pharmaceuticals',
+  category: 'Manufacturers',
+};
+
+testData.form = {
+  q1: 'Drug Registration',
+  q2: 'A',
+  q3: undefined,
+  q4: 'Panadol',
+};
+
+testData.application = {
+  id: 1,
+  name: 'Drug Registration',
+  status: 'Draft',
+  stage: 1,
+};
+
+testData.singleUserProperty = {
+  operator: 'objectProperties',
+  children: [
+    {
+      value: { object: 'user', property: 'firstName' },
+    },
+  ],
+};
+
+// SQL operator
+
+testData.getApplicationName = {
+  type: 'string',
+  operator: 'pgSQL',
+  children: [
+    {
+      value: 'SELECT name FROM application WHERE template_id = $1',
+    },
+    {
+      value: 2,
+    },
+  ],
+};
+
+testData.getListOfTemplates = {
+  type: 'array',
+  operator: 'pgSQL',
+  children: [
+    {
+      value: 'SELECT template_name FROM template',
+    },
+  ],
+};
+
+testData.countTemplates = {
+  type: 'number',
+  operator: 'pgSQL',
+  children: [
+    {
+      value: 'SELECT count(*) FROM template',
+    },
+  ],
+};
+
+testData.getListOfTemplates_noType = {
+  operator: 'pgSQL',
+  children: [
+    {
+      value: 'SELECT template_name FROM template',
+    },
+  ],
+};
+
+// GraphQL operator -- TO DO
+
+// More complex combinations
+
+testData.concatFirstAndLastNames = {
+  type: 'string',
+  operator: 'CONCAT',
+  children: [
+    {
+      operator: 'objectProperties',
+      children: [{ value: { object: 'user', property: 'firstName' } }],
+    },
+    {
+      value: ' ',
+    },
+    {
+      operator: 'objectProperties',
+      children: [{ value: { object: 'user', property: 'lastName' } }],
+    },
+  ],
+};
+
+testData.complex1 = {
+  operator: 'AND',
+  children: [
+    {
+      operator: '=',
+      children: [
+        {
+          operator: 'objectProperties',
+          children: [
+            {
+              value: { object: 'form', property: 'q1' },
+            },
+          ],
+        },
+        {
+          value: 'Drug Registration',
+        },
+      ],
+    },
+    {
+      operator: '!=',
+      children: [
+        {
+          type: 'number',
+          operator: 'pgSQL',
+          children: [
+            { value: 'SELECT COUNT(*) FROM user_organisation WHERE user_id = $1' },
+            {
+              operator: 'objectProperties',
+              children: [{ value: { object: 'user', property: 'id' } }],
+            },
+            {
+              operator: 'objectProperties',
+              children: [{ value: { object: 'organisation', property: 'id' } }],
+            },
+          ],
+        },
+        {
+          type: 'number',
+          value: 0,
+        },
+      ],
+    },
+  ],
+};
+
+testData.complex2 = {
+  operator: 'AND',
+  children: [
+    {
+      operator: '=',
+      children: [
+        {
+          operator: 'objectProperties',
+          children: [{ value: { object: 'application', property: 'stage' } }],
+        },
+        {
+          value: 1,
+        },
```

Comment:
Hmmm I think this one can be more complex. It seems to me that the `application.stage` property might be an object instead, with name and order (1). Something like that. In that case, this query might need a little tweak.

Reply:
Yes, true, but I've specified in my spec for this module that the `objectProperties` operator is still under consideration and we'll have a better idea of its requirements once we know how local state is being represented.